lib/systems/src/TorqueControllers.cpp

```diff
@@ -153,5 +153,104 @@ void TorqueControllerLoadCellVectoring::tick(
+        max_speed = std::max(max_speed, abs(wheel_rpms[i]));
+    }
+
+    writeout_.ready = true;
+
+    switch(launch_state){
+        case LaunchStates_e::LAUNCH_NOT_READY:
+            // set torques and speed to 0
+            writeout_.command.speeds_rpm[FL] = 0.0;
+            writeout_.command.speeds_rpm[FR] = 0.0;
+            writeout_.command.speeds_rpm[RL] = 0.0;
+            writeout_.command.speeds_rpm[RR] = 0.0;
+
+            writeout_.command.torqueSetpoints[FL] = brake_torque_req;
+            writeout_.command.torqueSetpoints[FR] = brake_torque_req;
+            writeout_.command.torqueSetpoints[RL] = brake_torque_req;
+            writeout_.command.torqueSetpoints[RR] = brake_torque_req;
+
+            //init launch vars
+            launch_speed_target = 0;
+            time_of_launch = tick.millis;
+
+            // check speed is 0 and pedals not pressed
+            if(pedalsData.accelPercent < launch_ready_accel_threshold
+               && pedalsData.brakePercent < launch_ready_brake_threshold
+               && max_speed * RPM_TO_METERS_PER_SECOND < launch_ready_speed_threshold)
+            {
+                launch_state = LaunchStates_e::LAUNCH_READY;
+            }
+
+            break;
+        case LaunchStates_e::LAUNCH_READY:
+            // set torques and speed to 0
+            writeout_.command.speeds_rpm[FL] = 0.0;
+            writeout_.command.speeds_rpm[FR] = 0.0;
+            writeout_.command.speeds_rpm[RL] = 0.0;
+            writeout_.command.speeds_rpm[RR] = 0.0;
+
+            writeout_.command.torqueSetpoints[FL] = brake_torque_req;
+            writeout_.command.torqueSetpoints[FR] = brake_torque_req;
+            writeout_.command.torqueSetpoints[RL] = brake_torque_req;
+            writeout_.command.torqueSetpoints[RR] = brake_torque_req;
+
+            //init launch vars
+            launch_speed_target = 0;
+            time_of_launch = tick.millis;
+
+            //check speed is 0 and brake not pressed
+            if (pedalsData.brakePercent >= launch_ready_brake_threshold
+                || max_speed >= launch_ready_speed_threshold)
+            {
+                launch_state = LaunchStates_e::LAUNCH_NOT_READY;
+            } else if(pedalsData.accelPercent >= launch_go_accel_threshold){
+                launch_state = LaunchStates_e::LAUNCHING;
+            }
+
+            //check accel above launch threshold and launch
+            break;
+        case LaunchStates_e::LAUNCHING:
+            //check accel below launch threshold and brake above
+            if(pedalsData.accelPercent <= launch_ready_accel_threshold
+               || pedalsData.brakePercent >= launch_ready_brake_threshold)
+            {
+                launch_state = LaunchStates_e::LAUNCH_NOT_READY;
+            }
+
+            launch_speed_target = (int16_t)((float) (tick.millis - time_of_launch) / 1000.0 * launch_rate_target_ * 60.0 / 1.2767432544 * 11.86);
+            launch_speed_target += 1500;
```

Comment:
more comments pls. Intent of this code is not clear.

Reply:
This was pretty much completely copied from HT07, Liwei helped clarify this and I added comments to show how this algo works and removed magic numbers that we already have defines for.